frontend/src/pages/submission/[id].jsx

```diff
@@ -0,0 +1,53 @@
+import React, { useEffect, useState } from 'react';
+import { useRouter } from 'next/router';
+import { Paper, Typography, TextField } from '@mui/material';
+
+const submission_service_port = 5433;
+
+export default function SubmissionPage() {
+  const router = useRouter();
+  // Get the dynamic part of the URL
+  const { id } = router.query;
+
+  const [submission, setSubmission] = useState([]);
+
+  useEffect(() => {
+    const fetchSubmissionDetails = async () => {
+      try {
+        console.log("Submission ID: " + id);
+        const response = await fetch(`http://localhost:${submission_service_port}/submission/${id}`);
+        if (!response.ok)
+          throw new Error(`HTTP error! status: ${response.status}`);
+        const data = await response.json();
+        setSubmission(data);
+      } catch (error) {
+        console.error('Error fetching submission details:', error);
```

Comment:
I think it would be better to explicitly handle different error codes e.g. 404 if no such submission. Though I am ok with handling that myself since I'm more familiar with the frontend.

Reply:
I'll leave that part to you